tests/NClient.Providers/NClient.Providers.Mapping.HttpResponses.Tests/HttpResponseBuilderTest.cs

```diff
@@ -272,5 +272,59 @@ public async Task Build_FailureHttpResponse_HttpResponseWithDataAndError()
             ((HttpResponseWithError<BasicEntity, string>) actualResult!).StatusCode.Should().Be(httpResponseMessage.StatusCode);
             ((HttpResponseWithError<BasicEntity, string>) actualResult!).Error.Should().Be(expectedError);
         }
+        
+        [Test]
+        public async Task Build_FailureHttpResponse_HttpResponseWithDataAndError_Should_Deconstruct()
+        {
+            var httpRequestMessage = new HttpRequestMessage();
+            var httpResponseMessage = new HttpResponseMessage
+            {
+                Content = new StreamContent(new MemoryStream()),
+                StatusCode = HttpStatusCode.NotFound
+            };
+            var responseContext = new ResponseContext<HttpRequestMessage, HttpResponseMessage>(httpRequestMessage, httpResponseMessage);
+            const string expectedError = "error";
+            _serializerMockSetup.Returns(expectedError);
+
+            var actualResult = await _responseToHttpResponseMapper.MapAsync(
+                typeof(HttpResponseWithError<BasicEntity, string>), responseContext, _serializerMock.Object, CancellationToken.None);
+            
+            actualResult.Should().BeOfType<HttpResponseWithError<BasicEntity, string>>();
+            ((HttpResponseWithError<BasicEntity, string>) actualResult!).StatusCode.Should().Be(httpResponseMessage.StatusCode);
+            ((HttpResponseWithError<BasicEntity, string>) actualResult!).Error.Should().Be(expectedError);
+            
+            var (_, error, httpResponse) = (HttpResponseWithError<BasicEntity, string>) actualResult;
```

Comment:
There is no case where the correct successful response is returned

Reply:
Ok. Done.